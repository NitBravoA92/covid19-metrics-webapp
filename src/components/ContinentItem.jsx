import PropTypes from 'prop-types';
import { BsArrowRightCircle } from 'react-icons/bs';
import { NavLink } from 'react-router-dom';

const ContinentItem = ({
  image, name, cases,
}) => {
  const continentName = name.replace(/\s/g, '-');

  return (
    <NavLink to={`${continentName}`} className="continent-item child-element">
      <div className="icon-show-details">
        <BsArrowRightCircle />
      </div>
      <div className="image">
        <img src={image} alt={name} loading="lazy" className="opacity-5" />
      </div>
      <div className="details">
        <h3 className="title">{ name }</h3>
        <p className="stats">
          { cases }
          {' '}
          cases
        </p>
      </div>
    </NavLink>
  );
};

ContinentItem.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  cases: PropTypes.string.isRequired,
};

export default ContinentItem;
