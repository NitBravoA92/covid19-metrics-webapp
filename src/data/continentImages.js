import Africa from '../assets/images/Africa.png';
import Asia from '../assets/images/Asia.png';
import AustraliaOceania from '../assets/images/AustraliaOceania.png';
import Europe from '../assets/images/Europe.png';
import NorthAmerica from '../assets/images/NorthAmerica.png';
import SouthAmerica from '../assets/images/SouthAmerica.webp';
import NewContinentFake from '../assets/images/NewContinentFake.png';

const continentImages = {
  Africa,
  Asia,
  'Australia-Oceania': AustraliaOceania,
  Europe,
  'North America': NorthAmerica,
  'South America': SouthAmerica,
  'New Continent Fake': NewContinentFake,
};

export default continentImages;
