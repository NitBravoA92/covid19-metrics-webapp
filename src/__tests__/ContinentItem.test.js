import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent, screen } from '@testing-library/react';
import { MemoryRouter, BrowserRouter } from 'react-router-dom';
import continentImages from '../data/continentImages';
import ContinentItem from '../components/ContinentItem';

describe('The ContinentItem component', () => {
  test('should render correctly into the DOM', () => {
    const { continentItemComponent } = render(
      <MemoryRouter>
        <ContinentItem image={continentImages.Africa} name="Africa" cases="1000" />
      </MemoryRouter>,
    );
    expect(continentItemComponent).toMatchSnapshot();
  });

  test('should navigate to the second page when user clicked on Australia-Oceania continent item', () => {
    render(
      <BrowserRouter>
        <ContinentItem image={continentImages['Australia-Oceania']} name="Australia-Oceania" cases="1,345,890" />
      </BrowserRouter>,
    );
    const continentItemLink = document.querySelector('.continent-item');
    fireEvent.click(continentItemLink);
    expect(window.location.pathname).toBe('/Australia-Oceania');
  });

  test('should render the text: Europe and the number of cases', () => {
    render(
      <MemoryRouter>
        <ContinentItem image={continentImages.Europe} name="Europe" cases="20,105,600" />
      </MemoryRouter>,
    );
    const continentItemTitle = screen.getByText(/Europe/i);
    const continentItemCases = screen.getByText(/20,105,600 cases/i);

    expect(continentItemTitle).toBeInTheDocument();
    expect(continentItemCases).toBeInTheDocument();
  });
});
