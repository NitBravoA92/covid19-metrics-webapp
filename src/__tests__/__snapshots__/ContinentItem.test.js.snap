// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ContinentItem component should render correctly into the DOM 1`] = `undefined`;
